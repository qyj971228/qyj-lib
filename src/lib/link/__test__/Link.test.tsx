import { create, act } from "react-test-renderer"
import { it, expect } from "@jest/globals"

import Link from "../Link"

it("Link Hover SnapShot", () => {
  // create
  const component = create(<Link page="http://www.facebook.com">Facebook</Link>)
  let json = component.toJSON()
  expect(json).toMatchSnapshot()

  // update
  act(() => {
    component.update(<Link page="http://www.google.com">Google</Link>)
  })
  json = component.toJSON()
  expect(json).toMatchSnapshot()

  // mouse enter
  act(() => {
    // @ts-ignore
    json.props.onMouseEnter()
  })
  json = component.toJSON()
  expect(json).toMatchSnapshot()

  // mouse leave
  act(() => {
    // @ts-ignore
    json.props.onMouseLeave()
  })
  json = component.toJSON()
  expect(json).toMatchSnapshot()
})

it("Link Hover", () => {
  // create
  const component = create(<Link page="http://www.facebook.com">Facebook</Link>)
  const instance = component.root
  let json = component.toJSON()

  // @ts-ignore
  expect(json.props.className).toBe('normal')
  expect(instance.findByProps({ className: "normal" }).props.className).toBe('normal')

  // update
  act(() => {
    component.update(<Link>Google</Link>)
  })
  json = component.toJSON()

  // @ts-ignore
  expect(json.children).toEqual(['Google'])

  // mouse enter
  act(() => {
    // @ts-ignore
    json.props.onMouseEnter()
  })
  json = component.toJSON()

  // @ts-ignore
  expect(json.props.className).toBe('hovered')

  // mouse leave
  act(() => {
    // @ts-ignore
    json.props.onMouseLeave()
  })
  json = component.toJSON()

  // @ts-ignore
  expect(json.props.className).toBe('normal')
})

it("default page is '#'", () => {
  const component = create(<Link>Facebook</Link>)
  let json = component.toJSON()
  // @ts-ignore
  expect(json.props.href).toBe('#')
})

it("did mouse click worked", () => {
  const component = create(<Link>Facebook</Link>)
  let json = component.toJSON()
  act(() => {
    // @ts-ignore
    json.props.onClick()
  })
})