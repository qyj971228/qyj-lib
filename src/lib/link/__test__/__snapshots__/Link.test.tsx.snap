// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Hover SnapShot 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link Hover SnapShot 2`] = `
<a
  className="normal"
  href="http://www.google.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link Hover SnapShot 3`] = `
<a
  className="hovered"
  href="http://www.google.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link Hover SnapShot 4`] = `
<a
  className="normal"
  href="http://www.google.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;
